src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -821,6 +821,18 @@ def load_adal_token_cache(self):
             self._load_service_principal_creds(all_entries)
             real_token = [x for x in all_entries if x not in self._service_principal_creds]
             self._adal_token_cache_attr = adal.TokenCache(json.dumps(real_token))
+
+            global _CLIENT_ID  # pylint: disable=global-statement
```

Comment:
Need a different approach than a global variable.

Reply:
One possible approach would be moving the client id into creds cache.  It seems to be the right place for it, SubscriptionFinder can use it from there, and the only other user is Profile which uses it for the cloud console.  This can continue to use it as well.

Reply:
Hi @DinoV probably best to set up a 30 minute meeting so we can talk through your requirements and consider the best solution moving forward.  We want to be very careful with changes to the core, especially something as critical as authentication.